From 3276b1a3e14478f524bccf43b212e6e9087b4ecd Mon Sep 17 00:00:00 2001
From: Stephane Danieau <stephane.danieau@st.com>
Date: Wed, 7 Mar 2018 11:49:18 +0100
Subject: [PATCH 08/37] pixman-renderer: support SHM I420 pixel format (debug
 purpose)

Pixman library only support YVU420 pixel format whereas I420 is equivalent
to YUV420.
Consequence is that I420 color will be wrong due to U and V planes
inversion in pixman pixel treatment but it will be sufficient for test
and debug purpose.

Change-Id: Id79b14e6074331e1f0150ccda6447d47d56bf6ef
Signed-off-by: Vincent Abriou <vincent.abriou@st.com>
---
 libweston/pixman-renderer.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libweston/pixman-renderer.c b/libweston/pixman-renderer.c
index d84104e..ee391dd 100644
--- a/libweston/pixman-renderer.c
+++ b/libweston/pixman-renderer.c
@@ -654,6 +654,10 @@ pixman_renderer_attach(struct weston_surface *es, struct weston_buffer *buffer)
 		case WL_SHM_FORMAT_NV12:
 			pixman_format = PIXMAN_nv12;
 			break;
+		case WL_SHM_FORMAT_YUV420:
+			pixman_format = PIXMAN_yv12;
+			weston_log("Warning: YUV420 will be treated as YVU420 by pixman. Color will be wrong!\n");
+			break;
 		default:
 			weston_log("Unsupported SHM buffer format\n");
 			weston_buffer_reference(&ps->buffer_ref, NULL);
@@ -1021,6 +1025,7 @@ pixman_renderer_init(struct weston_compositor *ec)
 	wl_display_add_shm_format(ec->wl_display, WL_SHM_FORMAT_BGR888);
 	wl_display_add_shm_format(ec->wl_display, WL_SHM_FORMAT_RGB565);
 	wl_display_add_shm_format(ec->wl_display, WL_SHM_FORMAT_NV12);
+	wl_display_add_shm_format(ec->wl_display, WL_SHM_FORMAT_YUV420);
 
 	wl_signal_init(&renderer->destroy_signal);
 
-- 
1.9.1

